src/python/pants/backend/scala/bsp/rules.py

```diff
@@ -138,10 +173,59 @@ async def bsp_scalac_options_request(request: ScalacOptionsParams) -> ScalacOpti
     return ScalacOptionsResult(items=tuple(result.item for result in results))
 
 
+# -----------------------------------------------------------------------------------------------
+# Compile Request
+# -----------------------------------------------------------------------------------------------
+
+
+@dataclass(frozen=True)
+class ScalaBSPCompileFieldSet(BSPCompileFieldSet):
+    required_fields = (ScalaSourceField,)
+    source: ScalaSourceField
+
+
+@rule
+async def bsp_scala_compile_request(
+    request: ScalaBSPCompileFieldSet,
+    union_membership: UnionMembership,
+    unzip: UnzipBinary,
+    bash: BashBinary,
+) -> BSPCompileResult:
+    coarsened_targets = await Get(CoarsenedTargets, Addresses([request.source.address]))
+    assert len(coarsened_targets) == 1
+    coarsened_target = coarsened_targets[0]
+    resolve = await Get(CoursierResolveKey, CoarsenedTargets([coarsened_target]))
+
+    result = await Get(
+        FallibleClasspathEntry,
+        ClasspathEntryRequest,
+        ClasspathEntryRequest.for_targets(
+            union_membership, component=coarsened_target, resolve=resolve
+        ),
+    )
+    _logger.info(f"scala compile result = {result}")
```

Comment:
The `level` for `FallibleClasspathEntry` was reduced to `DEBUG` [for successful compiles](https://github.com/pantsbuild/pants/blob/5db4b72f2cda68289204f0b2ebbe4e681df99914/src/python/pants/jvm/compile.py#L297-L298) a while back... that was fine for Go, but I don't think that it's necessarily right for the JVM. If we switched that back to `INFO`, we'd get more of this by default.

Reply:
For now, I've been leaving in this sort of debug logging in the BSP code since it is under active development. It'll be deleted once BSP support stabilizes.